src/Compilers/CSharp/Test/Syntax/Parsing/IgnoredDirectiveParsingTests.cs

```diff
@@ -336,6 +336,101 @@ public void AfterIf()
         EOF();
     }
 
+    [Fact]
+    public void AfterComment()
+    {
+        var source = """
+            #:x
+            // comment
```

Comment:
When it comes to the code "cracking" these directives without doing a full parse, we will have to be very mindful of comments, especially `/*` comments over multiple lines. It seems very plausible that users will want to "comment out" a directive such as `#:property`, which is something they can already do with `#define`s etc. I think SDK is going to be a more appropriate layer to add most of that testing. Just something to keep in mind.

Reply:
Yes, I'm planning to use Roslyn's lexer (via RoslynTokenParser experimental API) when this PR is merged and flows into the SDK.